passes/techmap/zinit.cc

```diff
@@ -139,11 +139,26 @@ struct ZinitPass : public Pass {
 
 				cell->setPort(ID(D), sig_d);
 				cell->setPort(ID(Q), initwire);
+
+				/*if (cell->type.in(ID($_DFFSR_NNN_), ID($_DFFSR_NNP_), ID($_DFFSR_NPN_), ID($_DFFSR_NPP_),
+							ID($_DFFSR_PNN_), ID($_DFFSR_PNP_), ID($_DFFSR_PPN_), ID($_DFFSR_PPP_)))
+				{
+					// TODO: I think I need a $_DFFRS_* cell where S has priority over R...
+					std::swap(cell->connections_.at(ID(R)), cell->connections_.at(ID(S)));
+				}
+				else*/ if (cell->type.in(ID($_DFF_NN0_), ID($_DFF_NN1_), ID($_DFF_NP0_), ID($_DFF_NP1_),
+							ID($_DFF_PN0_), ID($_DFF_PN1_), ID($_DFF_PP0_), ID($_DFF_PP1_)))
+				{
+					std::string t = cell->type.str();
+					t[8] = (t[8] == '0' ? '1' : '0');
+					cell->type = t;
+				}
 			}
 
-			for (auto &it : initbits)
-				if (donebits.count(it.first) == 0)
-					log_error("Failed to handle init bit %s = %s.\n", log_signal(it.first), log_signal(it.second));
+			if (!design->selected_whole_module(module))
```

Comment:
Why this if?  Why are we only reporting failure to handle init for partial selections?

Reply:
I'm not sure why this pass was so aggressive in firing an error if ANY `(* init *)` entry was left behind, but rather than thinking much about it I stuck it behind an `if` so that it won't throw any errors when operating on entire modules, to keep forward momentum.